### ISSUE TITLE
Arbiter does not join when redirecting savedatafolder

### ISSUE THREAD

**Issue by JyeGuru** [rwmt/Multiplayer#124]
Reproduced on vanilla Rimword+Royalty 1.2.2753 (both Steam and non-Steam) on Windows 10 20H2 with only Harmony and Multiplayer (both latest versions) enabled.

Steps to reproduce:
- Reset Rimworld to vanilla, and enable only Harmony and Multiplayer (to ensure no mod conflicts)
- Run Rimworld with `-savedatafolder=Local`
- Start a new colony (Any settings, just get to the ingame map)
- Host a server: LAN + Run Arbiter
- Notice that The Arbiter never joins the game

The `abiter_log.txt` shows the following relevant lines:

```
Command line arguments: -batchmode -nographics -arbiter -logfile arbiter_log.txt -connect=127.0.0.1:55054 -savedatafolder="Local"
RimWorld 1.2.2753 rev705
Save data folder overridden to "Local"
Critical error in root Start(): System.ArgumentException: Illegal characters in path.
  at System.IO.DirectoryInfo.CheckPath (System.String path) [0x0002f] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.IO.DirectoryInfo..ctor (System.String path, System.Boolean simpleOriginalPath) [0x00006] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.IO.DirectoryInfo..ctor (System.String path) [0x00000] in <567df3e0919241ba98db88bec4c6696f>:0 
  at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo..ctor(string)
  at Verse.GenFilePaths.get_SaveDataFolderPath () [0x000ec] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.GenFilePaths.FolderUnderSaveData (System.String folderName) [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.GenFilePaths.get_ConfigFolderPath () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.GenFilePaths.get_PrefsFilePath () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Prefs.Init () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Root.CheckGlobalInit () [0x0004f] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Root.Start () [0x0000a] in <6326ac03f61f45efb3178f3bfd084139>:0 
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

Critical error in root Start(): System.ArgumentException: Illegal characters in path.
  at System.IO.Path.Combine (System.String path1, System.String path2) [0x0003e] in <567df3e0919241ba98db88bec4c6696f>:0 
  at Verse.GenFilePaths.FolderUnderSaveData (System.String folderName) [0x00005] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.GenFilePaths.get_SavedGamesFolderPath () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.GenFilePaths.get_AllSavedGameFiles () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.SaveGameFilesUtility.GetAutostartSaveFile () [0x00009] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Root_Entry.Start () [0x0001e] in <6326ac03f61f45efb3178f3bfd084139>:0 
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)
```

This is then followed by a the following entries endlessly repeating forever:
```
Root level exception in Update(): System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.MusicManagerEntry.StartPlaying () [0x000a1] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at RimWorld.MusicManagerEntry.MusicManagerEntryUpdate () [0x0001b] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Root_Entry.Update () [0x00017] in <6326ac03f61f45efb3178f3bfd084139>:0 
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

Root level exception in Update(): System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.ResolutionUtility.Update () [0x0001f] in <6326ac03f61f45efb3178f3bfd084139>:0 
  at Verse.Root.Update () [0x00000] in <6326ac03f61f45efb3178f3bfd084139>:0 
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)
```

Have confirmed that The Arbiter works perfectly if the savedatafolder is not set, however this means it is not possible to have a "clean portable multiplayer copy" in order to keep config/mods/etc clean and always matching.

I have not been able to determine if this is an issue with the Rimworld core, as it only affects The Arbiter. The actual client loads from the specified folder just fine and multiplayer connects and works.

**Comment by JyeGuru** [comment]
https://github.com/rwmt/Multiplayer/blob/7aac8b54727d8626ec39429b97225fc88c807ab8/Source/Client/ClientNetworking.cs#L237

I think the issue is here - this has changed to require either no quotes, or quotes around the entire string, per the current Readme file in the game folder, referenced [here on the forums](https://ludeon.com/forums/index.php?topic=50413.0) and in [the 1.1.2560 patch notes](https://store.steampowered.com/news/app/294100/view/1705113791256438256)

```
OVERRIDING:
You can override the save data folder. This is useful, for example, if you want to install the game on a USB stick so you can plug and play it from anywhere. To do this, add this to the end of the command line used to launch the game:

    -savedatafolder=C:/Path/To/The/Folder

So it'll look something like this:

    C:/RimWorld/RimWorld.exe -savedatafolder=C:/Path/To/The/Folder

If you don't start the path with anything, it'll be relative to the game's root folder. So you could do this, to have the game save data in a folder called SaveData in its own root folder:

    -savedatafolder=SaveData

If there are spaces in the path, you'll need to surround the command with quotes:

    "-savedatafolder=Save Data"
```

If this is what I think it is, the fix should be super simple, changing that line to read this instead:
```
args += $" \"-savedatafolder={saveDataFolder}\"";
```

**Comment by JyeGuru** [comment]
Confirmed - I wrote [a quick little wrapper script](https://gist.github.com/JyeGuru/710e1eb2bf7c08d9dd43446b638fe090) to:
- Launch the game
- Wait for the arbiter process
- Copy the command line from it
- Kill it
- Replace the `-savedatafolder` argument with the quoting as suggested above
- Restart it
This works like a charm, The Arbiter is joining my local-config game.

**Comment by JyeGuru** [comment]
PR sent for this quick fix, as I noticed using a wrapper to re-launch caused some weirdness with The Arbiter process not correctly exiting when ending a session.

I've also noticed there's some interesting handling of the mod configs when running with a relative path for `-savedatafolder` but I haven't been able to narrow this down yet. Replacing the relative path with the full path to the same folder works correctly as expected. Getting some screenshots for this shortly.
